### ISSUE TITLE
FCMPlugin onNotification don't work 

### ISSUE THREAD

**Issue by junivillegas** [fechanique/cordova-plugin-fcm#372]
Hi i try to use the function onNotification but it don't work

My function: 
`FCMPlugin.onNotification(function(data) {
            if (data.wasTapped) {
                //Notification was received on device tray and tapped by the user.
                cordova.plugins.notification.badge.set(1);
                alert(JSON.stringify(data));
            } else {
                //Notification was received in foreground. Maybe the user needs to be notified.
                cordova.plugins.notification.badge.set(2);
                alert(JSON.stringify(data));
            }
            /*
            var temp = parseInt(data.badge);
            if (temp > 0) {
                cordova.plugins.notification.badge.set(temp);
            }
            */
        });`

All File index.js 

`var app = {
    // Application Constructor
    initialize: function() {
        document.addEventListener('deviceready', this.onDeviceReady.bind(this), false);

    },

    // deviceready Event Handler
    //
    // Bind any cordova events here. Common events are:
    // 'pause', 'resume', etc.
    onDeviceReady: function() {
        this.receivedEvent('deviceready');

        document.addEventListener('pause', function(event) {
            //window.alert('Pause');

        }, false);
        document.addEventListener('resume', function(event) {
            //window.alert('Resume');
        }, false);
        document.addEventListener('backbutton', function(event) {
            //window.alert('Back');
        }, false);
    },

    // Update DOM on a Received Event
    receivedEvent: function(id) {

        cordova.plugins.notification.badge.requestPermission(function(granted) {});

        cordova.plugins.notification.badge.configure({ autoClear: false });

        pushNotification();
    },

    pushNotification: function() {

        FCMPlugin.onNotification(function(data) {

            if (data.wasTapped) {
                //Notification was received on device tray and tapped by the user.
                cordova.plugins.notification.badge.set(1);
                alert(JSON.stringify(data));
            } else {
                //Notification was received in foreground. Maybe the user needs to be notified.
                cordova.plugins.notification.badge.set(2);
                alert(JSON.stringify(data));
            }

            /*
            var temp = parseInt(data.badge);
            if (temp > 0) {
                cordova.plugins.notification.badge.set(temp);
            }
            */
        });

    }
};

app.initialize();`

Ajax call 

`$.ajax({
                    type: 'POST',
                    url: 'https://fcm.googleapis.com/fcm/send',
                    headers: {
                        Authorization: 'key=' + 'KEY'
                    },
                    contentType: 'application/json',
                    dataType: 'json',
                    data: JSON.stringify({
                        'to': '/topics/all',
                        'priority':'high',
                        'content_available': true,
                        'notification': {
                            'title' : 'Test',
                            'body' : 'Test',
                            'click_action': 'FCM_PLUGIN_ACTIVITY',
                            'badge' : '8' 
                        },
                        'data': {
                            'useBadge' : true,
                            'badge': '8'
                        }
                    }),
                    success: function(response) {
                        console.log(response);
                    },
                    error: function(xhr, status, error) {
                        console.log(xhr.error);
                    }
                });`

Any idea how could be the error?

Other plugin https://github.com/katzer/cordova-plugin-badge

Android: 7 in Samsung s8

regards

**Comment by Alb93** [comment]
Do you receive the notification also when the app is killed?

**Comment by junivillegas** [comment]
Yes I recibe the notification also when the app is killed, but never enter in the function onNotification

**Comment by akshaysm** [comment]
hello i am facing a similar problem in foreground, notification is caught but data is not present. any resolution?

**Comment by Alb93** [comment]
@junivillegas have you installed some other plugins for the push notificatomi or only the fcm one?

And in which model phone do you receive the killed-app notifications?

**Comment by prantikv** [comment]
I have the opposite issue. I get the data when app is in the foreground but do not get data when app is in the background.

I am getting the notification though

**Comment by cipriancirstea** [comment]
I have a similar problem. When app in background the notification arrives, but it doesn't enter the if(data.wasTapped) {}.

**Comment by djpexhibit** [comment]
I'm having the same issue. Notification receive when app got killed but not come to onNotificationOpen. Also not when it is in foreground.

**Comment by ourarash** [comment]
Any solutions for this issue?

**Comment by jamiecounsell** [comment]
@Alb93 good call. My issue was I also had `phonegap-plugin-push` installed. Uninstalling it caused the `onNotification` to fire as expected.

**Comment by demaniak** [comment]
*sigh* unfortunately we did NOT have `phonegap-plugin-push` installed, but we still can't get anything while app is in the foreground (android). if app is backgrounded, all is 100% fine however.

**Comment by demaniak** [comment]
Thank you @jamiecounsell ! 

There was some extreme weirdness in our case: the `cordova/plugin/cordova-plugin-push` folders existed, but were all empty. Initially leading us to believe the plugin was NOT installed.

However, some `adb logcat` use on a running application gave the hint that it might actually still be active somehow. Long story short, apparently the plugin was still alive in  `platforms/android`. after removing the android platform, and re-adding it...the `cordova/plugin/cordova-plugin-push` was well and truly gone, and this plugin started behaving as it should!
